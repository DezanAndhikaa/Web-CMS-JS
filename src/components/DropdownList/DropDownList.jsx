import React from 'react'
import { Select, MenuItem, FormLabel, Menu } from '@material-ui/core'
import './DropDownList.scss'

class DropDownList extends React.Component{
    state = {
        pagesize: 0,
    }
    // handleClick = async(prevState, value) => {
        // const { myValue } = event.currentTarget.dataset
        // this.setState({pagesize : numberOfPages})
        // await this.setState({ pagesize : value.target.value});
        // await this.props.clearSelectedSalesPlans(), 
        // await this.props.clearSelectedServicePlans(),
        //     console.log('this is pagenumber size data', prevState.target.value)
        //     console.log('this is pagenumber size data', this.state.pagesize)

        // this.props.onPageSize(value)
        // await this.props.clearSelectedServicePlans();
        // await this.props.clearSelectedSalesPlans();
        // await this.props.updateServiceParameter({ ...this.props.serviceParameter.dataFilter, PageSize: this.state.pagesize})
    // }

    handleChange = async(event) => {
        this.setState(
            { pagesize : event.target.value},
            // () => this.props.updateSalesParameter({ ...this.props.salesParameter.dataFilter, PageSize: this.state.pagesize}), 
            () => this.props.handleClickShowPerPage(this.state.pagesize)
            );
        // await this.handleClick(event.target.value)
        // this.handleClick(event.target.value)
        // await this.props.updateSalesParameter({ ...this.props.salesParameter.dataFilter, PageSize: this.state.pagesize})
        // await this.props.updateServiceParameter({ ...this.props.serviceParameter.dataFilter, PageSize: this.state.pagesize})
      }
	render(){
		return(
			<div className="show-per-page">
				<span>
					<FormLabel className="label">Show</FormLabel>
					<Select variant="outlined" 
						className="dropdown-list" 
                        labelId="label" id="select" 
                        // value={this.state.pagesize}
                        defaultValue={2}
                        onChange={this.handleChange}
                        // onClick={this.handleClick}
					>
						<MenuItem className="dropdown-list" value="2">2</MenuItem>
						<MenuItem className="dropdown-list" value="4">4</MenuItem>
						<MenuItem className="dropdown-list" value="6">6</MenuItem>
						<MenuItem className="dropdown-list" value="100">100</MenuItem>
						<MenuItem className="dropdown-list" value="100000">All</MenuItem>
					</Select>
					<FormLabel className="label">entries</FormLabel>
				</span>
			</div>
		)
	}
}

export default DropDownList;