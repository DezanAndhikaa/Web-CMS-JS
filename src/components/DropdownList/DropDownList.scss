@import '../../themes/functions';
@import '../../themes/variables';

.show-per-page{
    display: flex;
    flex-direction: row;
    .dropdown-list{
        margin: 0 vw(10) 0 vw(10);
        background-color: white;
        box-shadow: 0px vw-tab(2) vw-tab(4) rgba(0, 0, 0, 0.161);
        width: vw(100);
        min-width: vw(100);
        height: vw(35.3);
        font-size: vw(16);
        font-weight: bold;
        line-height: vw(16) !important;
        font-family: 'Lato-Regular';
        padding: 0px 2px;
    }
    .label{
        font-size: vw(16);
        font-weight: bold;
        line-height: vw(16) !important;
        font-family: 'Lato-Regular';
    }
}