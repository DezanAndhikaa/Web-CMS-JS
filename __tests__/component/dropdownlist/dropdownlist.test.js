import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import DropdownList from '../../../src/components/DropdownList/DropDownList';

Enzyme.configure({adapter:new Adapter()});
it("render dropdown list correctly", () => {
    const wrapper = shallow (
        <DropdownList />
    )
    expect(wrapper).toMatchSnapshot();
})