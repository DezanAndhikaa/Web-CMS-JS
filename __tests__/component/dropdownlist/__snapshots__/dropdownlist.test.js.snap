exports[`render dropdown list correctly 1`] = `
<div
  className="show-per-page"
>
  <WithStyles(ForwardRef(FormLabel))
    className="label"
  >
    Show
  </WithStyles(ForwardRef(FormLabel))>
  <WithStyles(ForwardRef(Select))
    className="dropdown-list"
    defaultValue={10}
    id="select"
    labelId="label"
    onChange={[Function]}
    variant="outlined"
  >
    <WithStyles(ForwardRef(MenuItem))
      className="dropdown-list"
      value="5"
    >
      5
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      className="dropdown-list"
      value="10"
    >
      10
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      className="dropdown-list"
      value="15"
    >
      15
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      className="dropdown-list"
      value="20"
    >
      20
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      className="dropdown-list"
      value="25"
    >
      25
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
  <WithStyles(ForwardRef(FormLabel))
    className="label"
  >
    Entries
  </WithStyles(ForwardRef(FormLabel))>
</div>
`;